webapp/graphite/tags/utils.py

```diff
@@ -0,0 +1,94 @@
+"""Utility functions for tag databases."""
+import abc
+
+try:
+    from importlib import import_module
+except ImportError:  # python < 2.7 compatibility
+    from django.utils.importlib import import_module
+
+
+class TaggedSeries(object):
+  @classmethod
+  def parse(cls, path):
+    segments = path.split(';')
+
+    metric = segments[0]
+    if not metric:
+      raise Exception('Cannot parse path %s, no metric found' % path)
+
+    tags = {}
+
+    for segment in segments[1:]:
+      tag = segment.split('=', 1)
+      if len(tag) != 2 or not tag[0]:
```

Comment:
Yeah, I'm still not 100% sure how we want to handle cases like empty values, and it opens up some questions about negative matches.  The [prometheus docs](https://prometheus.io/docs/querying/basics/#instant-vector-selectors) outline their rules:
```
Label matchers that match empty label values also select all time series that do not have the specific label set at all. Regex-matches are fully anchored.

Vector selectors must either specify a name or at least one label matcher that does not match the empty string. 
```

This is definitely an area we need to better define.